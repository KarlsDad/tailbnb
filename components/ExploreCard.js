export default ({ img, title }) => {
  return (
    <div className='lg:w-1/3'>
      <div className='mr-4 flex flex-wrap items-center justify-start shadow-md rounded-xl bg-white'>
        <img
          src={img}
          className='rounded-xl rounded-r-none'
          style={{ height: 80, width: 'auto' }}
          alt='explore1'
        />
        <p class='pl-4 text-gray-750 font-semibold'>{title}</p>
      </div>
    </div>
  );
};
