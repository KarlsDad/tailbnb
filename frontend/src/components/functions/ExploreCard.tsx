import * as React from 'react';
import Link from 'next/link';

interface Prop {
  img: string;
  title: string;
}

export const ExploreCard: React.FC<Prop> = ({ img, title }) => {
  return (
    <div className='w-30/31'>
      <Link
        href={`/${title.toLowerCase()}/index.tsx`}
        as={`/${title.toLowerCase()}`}>
        <div className='h-full w-32 sm:w-full flex flex-wrap items-center justify-start shadow-md rounded-xl bg-white'>
          <img
            src={img}
            className='h-24 w-full md:h-full xl:h-20 rounded-b-none xl:h-full xl:w-32 rounded-lg xl:rounded-xl xl:rounded-r-none'
          />
          <p className='text-sm py-3 xl:py-0 pl-4 sm:pl-4 text-gray-750 font-semibold'>
            {title}
          </p>
        </div>
      </Link>
    </div>
  );
};
