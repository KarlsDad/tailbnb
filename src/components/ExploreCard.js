import React from 'react';

export default ({ img, title }) => {
  return (
    <div className='w-30/31'>
      <div className='h-full w-32 sm:w-full flex flex-wrap items-center justify-start shadow-md rounded-xl bg-white'>
        <img
          src={img}
          className='h-24 w-full md:h-full xl:h-20 rounded-b-none xl:h-full xl:w-32 rounded-lg xl:rounded-xl xl:rounded-r-none'
          alt='explore1'
        />
        <p class='text-sm py-3 xl:py-0 pl-4 sm:pl-4 text-gray-750 font-semibold'>
          {title}
        </p>
      </div>
    </div>
  );
};
