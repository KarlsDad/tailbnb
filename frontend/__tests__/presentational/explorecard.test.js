import React from 'react';
import { shallow } from 'enzyme';

// Component
import ExploreCard from '../../src/components/presentational/ExploreCard';

describe('ExploreCard', () => {
  it('should render correctly in "debug" mode', () => {
    const title = 'random';
    const component = shallow(<ExploreCard title={title} />);
    expect(component).toMatchSnapshot();
  });
});
