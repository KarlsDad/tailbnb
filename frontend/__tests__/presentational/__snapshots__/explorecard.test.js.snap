// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreCard should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
